src/main/java/seedu/address/logic/parser/CommandArgument.java

```diff
@@ -1,26 +1,57 @@
 package seedu.address.logic.parser;
 
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PREAMBLE;
+
+import java.util.Optional;
+
 public class CommandArgument {
+    private final String name;
     private final Prefix prefix;
     private final boolean isRequired;
     private final boolean isMultiple;
 
-    private CommandArgument(Prefix prefix, boolean isRequired, boolean isMultiple) {
+    protected CommandArgument(String name, Prefix prefix, boolean isRequired, boolean isMultiple) {
+        assert (name != null || !prefix.equals(PREFIX_PREAMBLE));
+        this.name = name;
         this.prefix = prefix;
         this.isRequired = isRequired;
         this.isMultiple = isMultiple;
     }
 
     public static CommandArgument requiredSingle(Prefix prefix) {
-        return new CommandArgument(prefix, true, false);
+        return requiredSingle(prefix, null);
+    }
+
+    public static CommandArgument requiredSingle(Prefix prefix, String name) {
+        return new CommandArgument(name, prefix, true, false);
     }
 
     public static CommandArgument optionalSingle(Prefix prefix) {
-        return new CommandArgument(prefix, false, false);
+        return optionalSingle(prefix, null);
+    }
+
+    public static CommandArgument optionalSingle(Prefix prefix, String name) {
+        return new CommandArgument(name, prefix, false, false);
     }
 
     public static CommandArgument optionalMultiple(Prefix prefix) {
-        return new CommandArgument(prefix, false, true);
+        return optionalMultiple(prefix, null);
+    }
+
+    public static CommandArgument optionalMultiple(Prefix prefix, String name) {
+        return new CommandArgument(name, prefix, false, true);
+    }
+
+    public static CommandArgument unknown(Prefix prefix) {
+        return new CommandArgument(null, prefix, false, false);
+    }
```

Comment:
Is there a reason for this function? From what I can see `new CommandArgument(null, ...` will cause an assertion error from the `assert (name != null || !prefix.equals(PREFIX_PREAMBLE));` in the constructor (line 14). 

Given that assertions should never happen, does this break the usage of assertions? Especially since assertions can be disabled. Unless I'm missing something?

Reply:
This basically asserts that the preamble must always be named. As supposed to other prefixes like `tag` and `done`.
`tag/TAG` is a `tag/` prefix named `TAG`. `done/` is an unnamed `done/` prefix because there's not supposed to be an argument coming after it.